### PR TITLE
Feat: HW wallet parameter to set zero balance threshold

### PR DESCRIPTION
**Pull request by Adamj1232**
### Description
This PR adds an init parameter `consecutiveEmptyAccountThreshold` with type `number` to HW wallets that use Account Select (`trezor`, `keepkey` & `keystone`).
This parameter defaults to `5`
This can be changed by the dev to allow more or less consecutive empty addresses between addresses with balance when looping through the hardware wallet accounts.

Screenshot 2023-05-12 at 1 26 31 PM

### Checklist
- [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [x] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### CODE REVIEW

**Review by leightkt — APPROVED** [review]
I don't have a trezor wallet- so I can't test locally, but the code looks good!

File: packages/demo/src/App.svelte [hunk 1]
```diff
@@ -150,7 +150,9 @@
 
   const trezorOptions = {
     email: 'test@test.com',
-    appUrl: 'https://www.blocknative.com'
+    appUrl: 'https://www.blocknative.com',
+
```
Comment (Adamj1232) [inline packages/demo/src/App.svelte:154]:
```suggestion
```

File: packages/hw-common/src/validation.ts [hunk 2]
```diff
@@ -29,6 +29,7 @@ const selectAccountOptions = Joi.object({
   chains: chains,
   scanAccounts: Joi.function().arity(1).required(),
   supportsCustomPath: Joi.bool(),
+  consecutiveEmptyAccountThreshold: Joi.bool(),
```
Comment (aaronbarnardsound) [inline packages/hw-common/src/validation.ts:32]:
```suggestion
  consecutiveEmptyAccountThreshold: Joi.number(),
```
Should this be set to `number`?

**Review by aaronbarnardsound — APPROVED** [review]
Just a question around the validation type.

**Pull request merged**